doc/chapters/api/index.rst

```diff
@@ -3,131 +3,230 @@
 ---------
-Binds a path on the host to the container. Any missing parent directories in
-the path will be created.
-
-:Parameters:
-        :containerID: ``int32`` The ID obtained by CreateContainer method.
-        :pathInHost: ``string`` absolute path in the host.
-        :pathInContainer: ``string`` the absolute path to mount to in the container.
-        :readOnly: ``bool`` indicates whether the directory is read-only or not.
-
-|
+Binds a path on the host to the container. Any missing parent directories in the path will be
+created.
+
+Parameters
+##########
+* containerID: ``int32`` The ID obtained by CreateContainer method.
+* pathInHost: ``string`` absolute path in the host.
+* pathInContainer: ``string`` the absolute path to mount to in the container.
+* readOnly: ``bool`` indicates whether the directory is read-only or not.
+
+Error sources
+#############
+* Invalid ID: No matching container exists.
+* Bad state: The container is not started or ready
+* Bad path
+    * The host path does not exist
+    * The destination path is not absolute
+    * The destination path is already mounted to
+* Filesystem issues
+    * Could not create parent directories
+    * Could not create the target directory (if mounting a directory)
+    * Could not touch the target file (if mounting a file)
+* Mount issues
+    * Mount failed (:linuxman:`mount(2)` failed)
+    * Couldn't move the mount
+    * Failed to re-mount read-only
+
+Currently, failing mounts are not rolled back.
 
 Create
 ------
 Creates a container with given configuration.
 
-:Parameters:
-        :config: ``string`` config file in json format.
+Parameters
+##########
+* config: ``string`` config file in json format.
 
-                Example config JSON::
+Example config JSON::
 
-                [{"enableWriteBuffer": true}]
+[{"enableWriteBuffer": true}]
 
-|
+Return Values
+#############
+* containerID: ``int32`` ID of created SoftwareContainer.
 
-:Return Values:
-        :containerID: ``int32`` ID of created SoftwareContainer.
+Error sources
+#############
+* Network setup issues
+    * Bridge device not available and unable to set it up
+    * Given netmask can not be used to assign IP address to container
+    * See :ref:`Network setup<network-setup>`
 
-Destroy
--------
-Tears down all active gateways related to container and shuts down the
-container with all reserved sources.
+* LXC issues
+    * Couldn't prepare file system while initializing container
+    * Container ID already being in use (caused by lingering old containers)
+    * LXC config couldn't be loaded
+    * LXC could not create container
 
-:Parameters:
-        :containerID: ``int32`` The ID obtained by CreateContainer method.
+If the error about ID already being in use shows up, one can try to restart the
+softwarecontainer-agent, since it does a cleanup of old containers on startup.
 
-|
+Destroy
+-------
+Tears down all active gateways related to container and shuts down the container with all reserved
```

Comment:
"related to container" -> "related to the container"
You are missing a few "the" in the sentences. When it's for making the sentences brief (in the bullet lists I think it's ok).